source/validate_instruction.cpp

```diff
@@ -145,14 +146,51 @@ spv_result_t LimitCheckIdBound(ValidationState_t& _,
 // Checks that the number of OpTypeStruct members is within the limit.
 spv_result_t LimitCheckStruct(ValidationState_t& _,
                               const spv_parsed_instruction_t* inst) {
+  if (SpvOpTypeStruct != inst->opcode) {
+    return SPV_SUCCESS;
+  }
+
   // Number of members is the number of operands of the instruction minus 1.
   // One operand is the result ID.
   const uint16_t limit = 0x3fff;
-  if (SpvOpTypeStruct == inst->opcode && inst->num_operands - 1 > limit) {
+  if (inst->num_operands - 1 > limit) {
     return _.diag(SPV_ERROR_INVALID_BINARY)
            << "Number of OpTypeStruct members (" << inst->num_operands - 1
            << ") has exceeded the limit (" << limit << ").";
   }
+
+  // Section 2.17 of SPIRV Spec specifies that the "Structure Nesting Depth"
+  // must be less than or equal to 255.
+  // This is interpreted as structures including other structures as members.
+  // The code does not follow pointers or look into arrays to see if we reach a
+  // structure downstream.
+  //
+  // This lambda takes a valid struct ID and finds its nesting depth.
+  // The nesting depth of a struct is 1+(largest depth of any member).
+  // Scalars are at depth 0.
+  std::function<int(uint32_t)> find_struct_depth_recursive = [&](uint32_t id) {
+    int max_member_depth = 0;
+    auto cur_instr = _.FindDef(id);
+    // Struct members start at word 2 of OpTypeStruct instruction.
+    for (size_t word_i = 2; word_i < cur_instr->words().size(); ++word_i) {
```

Comment:
I don't see how/why to use `std::accumulate`?... I only need to do `1+max{d1,d2,d3,...}` where `di` is the depth of member i.